File: src/main/kotlin/onl/tesseract/srp/skill/adapter/serverside/jpa/repository/SkillResultCacheJpaRepository.java
package onl.tesseract.srp.skill.adapter.serverside.jpa.repository;

import onl.tesseract.srp.skill.adapter.serverside.jpa.entity.cache.CacheEntity;
import onl.tesseract.srp.skill.adapter.serverside.jpa.entity.PlayerSkillEmbeddable;
import org.springframework.data.jpa.repository.JpaRepository;
import org.springframework.stereotype.Repository;

import java.util.UUID;

@Repository
public interface SkillResultCacheJpaRepository extends JpaRepository<CacheEntity, Long> {

    void deleteByPlayerSkill(PlayerSkillEmbeddable playerSkill);

}

File: src/main/kotlin/onl/tesseract/srp/repository/hibernate/skill/SkillResultCacheJpaRepository.kt
package onl.tesseract.srp.repository.hibernate.skill

import org.springframework.data.jpa.repository.JpaRepository
import java.util.*

interface SkillResultCacheJpaRepository : JpaRepository<SkillResultCacheEntity, Long> {
    fun findByPlayerUuidAndSkillName(playerUuid: UUID, skillName: String): SkillResultCacheEntity?
    fun deleteByPlayerUuidAndSkillName(playerUuid: UUID, skillName: String)
}
